serve/graph/subscription.resolvers.go

```diff
@@ -0,0 +1,39 @@
+package graph
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+// Code generated by github.com/99designs/gqlgen version v0.17.45
+
+import (
+	"context"
+
+	"github.com/gnolang/tx-indexer/serve/graph/model"
+	"github.com/gnolang/tx-indexer/types"
+)
+
+// Transactions is the resolver for the transactions field.
+func (r *subscriptionResolver) Transactions(ctx context.Context, filter model.TransactionFilter) (<-chan *model.Transaction, error) {
+	return handleChannel(ctx, r.manager, func(nb *types.NewBlock, c chan<- *model.Transaction) {
+		for _, tx := range nb.Results {
+			transactionResolver := NewTransactionResolver(model.NewTransaction(tx))
```

Comment:
This is memory inefficient hitting badly to the Go GC. You are creating a new object per result obtained from the storage, only needed to apply a filter. WDYT about removing the TransactionResolver struct and the BlockResolver struct and just leaving filter methods? Example:

```go

func FilterTransactionBy(tx *model.Transaction, f model.TransactionFilter) bool {...}

```

Reply:
I removed the resolver and changed it to a function.